pkg/provisioningv2/rke2/planner/store.go

```diff
@@ -58,6 +58,25 @@ func onlyRKE(machines []*capi.Machine) (result []*capi.Machine) {
 	return
 }
 
+func (p *PlanStore) ClusterHasBeenDeliveredPlan(cluster *capi.Cluster, rkeControlPlane *rkev1.RKEControlPlane) (bool, error) {
+	plan, err := p.Load(cluster, rkeControlPlane)
+	if err != nil {
+		return false, err
+	}
+
+	if plan.Nodes == nil {
+		return false, nil
+	}
+
+	for _, e := range plan.Nodes {
+		if e.AppliedPlan != nil && e.InSync {
```

Comment:
I think in this case, we actually want to check to make sure that the initial plan isn't set. Even though the plan may not have completed, that doesn't necessarily mean that it won't complete in between the time we observe it and the time we act on it, or that there were other changes (i.e. plan may have applied but probes may not yet be complete). We may need to check the secret, convert the `Plan` field to a pointer, or add a function/field (or both) to indicate that a plan was delivered to the node via the secret..

Reply:
I'm assuming that once the plan secret has been generated it will always be available in the store? In which case we could, like you said, just query the secrets generated for all of the machines and check if there are any. If there are not, can we safely assume the initial secret has not been created yet? 

I've updated the last commit with a different function for checking if plans have been delivered, I still getting to know the planner code base but hopefully its closer to what we're looking for

Reply:
Yeah, any plan that has been delivered should work; however, there is probably an edge case when dealing with a cluster that is being "etcd restored" from nothing, because the initial plan will not actually net a functional cluster. Maybe we need to be a little more intelligent about how we determine to block... need to think about this